### ISSUE TITLE
Can’t detect Droid Assault

### ISSUE THREAD

**Issue by basxto** [MikeMaximus/gbm#100]
GBM can’t detect [Droid Assault](https://pcgamingwiki.com/wiki/Droid_Assault)

In [gamesPy](https://github.com/basxto/gamesPy/tree/09a4f3c49e4aa04a541a730cf89493f601a21810) I managed to detect it with process name `bash` and argument `droidassault.sh`. GBM can’t detect it with the same settings.

I only managed to detect it via [exe](https://psutil.readthedocs.io/en/latest/#psutil.Process.exe) and not via [name](https://psutil.readthedocs.io/en/latest/#psutil.Process.name) how I usually do it.

I don’t know whether .net allows to get the absolute path, but if it doesn’t work you could try to check argument[0], which should also be the applications name. (It _can_ be changed, though)

**Comment by MikeMaximus** [comment]
I'm guessing that it's because bash is only running long enough to execute the commands in droidassault.sh then terminating.  GBM won't confirm a detected process until it's been running at least 10 seconds, and only checks at 5 second intervals.

Does the droidassault.sh script fire off another binary to run to game?  It's the name of that binary that GBM will need to detect.

**Comment by basxto** [comment]
It does start java, but the game is not run as a background process.
This means the shell waits until the child process has finished.

`java32` `DroidAssault.jar` does work, do you just check whether the string is contained anywhere in the parameter strings?

But the script is definitely running, since I can track it with my gamesPy which scans all processes every 10 seconds and then waits for that process. GBM should be able to also detect that process.

My test gave me the following data:

```terminal
$ ps --pid 6941
  PID TTY          TIME CMD
 6941 ?        00:00:00 droidassault.sh
$ ps -F --pid 6941
UID        PID  PPID  C    SZ   RSS PSR STIME TTY          TIME CMD
dabascht  6941  6940  0 18107  7204   7 17:39 ?        00:00:00 /bin/bash /mnt/olhdd/steam/steamapps/common/Droid Assault/droidassault.sh
$ pstree 6941 -l
droidassault.sh───java32───33*[{java32}]
$ pstree 6941 -al
droidassault.sh /mnt/olhdd/steam/steamapps/common/Droid Assault/droidassault.sh
  └─java32 -server -Xbootclasspath/p:./lib/rt32.jar -Djava.library.path=/mnt/olhdd/steam/steamapps/common/Droid Assault -Dorg.lwjgl.librarypath=/mnt/olhdd/steam/steamapps/common/Droid Assault -Dorg.lwjgl.util.NoChecks=false -Djava.net.preferIPv4Stack=true -Dnet.puppygames.applet.Launcher.resources=resources-steam.dat -Dnet.puppygames.applet.Game.gameResource=game.steam -XX:MaxGCPauseMillis=3 -XX:+TieredCompilation -XX:Tier2CompileThreshold=70000 -XX:+DoEscapeAnalysis -Xms64m -Xmx256m -Xincgc -cp DroidAssault.jar:maps.jar:music.jar:fx-mono.jar:fx-stereo.jar:images.jar:gfx.jar:fonts.jar:data-steam.jar:common-cp-java6.jar:common.jar:spgl-lite.jar:lwjgl.jar:lwjgl_util.jar:gson.jar:jinput.jar:ogg.jar:gamecommerce.jar:steampuppy-public.jar:remote.jar:steampuppy.jar net.puppygames.applet.Launcher
      └─33*[{java32}]
``` 

I also modified gamesPy a bit print all processes, which contain droid or Droid in their name or parameters
```py
droidtest = re.compile('(d|D)roid');
if ((pinfo['name'] and droidtest.search(pinfo['name']))
 or (pinfo['exe'] and droidtest.search(pinfo['exe']))
 or (pinfo['cmdline'] and [arg for arg in pinfo['cmdline'] if droidtest.search(arg)])
):
    print('pid: {}\nname: {}\nexe: {}\ncmdline: {}\n---'.format(pinfo['pid'],pinfo['name'], pinfo['exe'], pinfo['cmdline']));
```
Which gave
```
pid: 6941
name: droidassault.sh
exe: /usr/bin/bash
cmdline: ['/bin/bash', '/mnt/olhdd/steam/steamapps/common/Droid Assault/droidassault.sh']
---
pid: 6945
name: java32
exe: /mnt/olhdd/steam/steamapps/common/Droid Assault/bin/java32
cmdline: ['./bin/java32', '-server', '-Xbootclasspath/p:./lib/rt32.jar', '-Djava.library.path=/mnt/olhdd/steam/steamapps/common/Droid Assault', '-Dorg.lwjgl.librarypath=/mnt/olhdd/steam/steamapps/common/Droid Assault', '-Dorg.lwjgl.util.NoChecks=false', '-Djava.net.preferIPv4Stack=true', '-Dnet.puppygames.applet.Launcher.resources=resources-steam.dat', '-Dnet.puppygames.applet.Game.gameResource=game.steam', '-XX:MaxGCPauseMillis=3', '-XX:+TieredCompilation', '-XX:Tier2CompileThreshold=70000', '-XX:+DoEscapeAnalysis', '-Xms64m', '-Xmx256m', '-Xincgc', '-cp', 'DroidAssault.jar:maps.jar:music.jar:fx-mono.jar:fx-stereo.jar:images.jar:gfx.jar:fonts.jar:data-steam.jar:common-cp-java6.jar:common.jar:spgl-lite.jar:lwjgl.jar:lwjgl_util.jar:gson.jar:jinput.jar:ogg.jar:gamecommerce.jar:steampuppy-public.jar:remote.jar:steampuppy.jar', 'net.puppygames.applet.Launcher']
---
```

But neither `droidassault.sh` nor `bash` `droidassault.sh` get detected.

**Comment by MikeMaximus** [comment]
This is caused by a bug that I introduced with the fix for issue #90.  When multiple copies of the same process are running (which is highly likely with bash) detection will probably fail, depending on the order of the processes.  I believe I got a solution, but I want to do some testing before a commit.

**Comment by MikeMaximus** [comment]
Let me know if the latest commit helps.  I missed your question before, yes GBM just checks if there's a match to any part of the parameter.
